src/custom/conversions.rs

```diff
@@ -0,0 +1,463 @@
+    fn build_trunc_op(
+        &mut self,
+        signed: bool,
+        log_width: u64,
+        args: EmitOpArgs<'c, ExtensionOp, H>,
+    ) -> Result<()> {
+        // Note: This logic is copied from `llvm_type` in the IntTypes
+        // extension. We need to have a common source of truth for this.
+        let (width, int_max_value_s, int_max_value_u) = match log_width {
+            0..=3 => Ok((8, i8::MAX as u64, u8::MAX as u64)),
+            4 => Ok((16, i16::MAX as u64, u16::MAX as u64)),
+            5 => Ok((32, i32::MAX as u64, u32::MAX as u64)),
+            6 => Ok((64, i64::MAX as u64, u64::MAX)),
+            m => Err(anyhow!(
+                "IntTypesCodegenExtension: unsupported log_width: {}",
+                m
+            )),
+        }?;
+
+        let hugr_int_ty = INT_TYPES[log_width as usize].clone();
+        let int_ty = self
+            .0
+            .typing_session()
+            .llvm_type(&hugr_int_ty)?
+            .into_int_type();
+
+        let hugr_sum_ty = sum_with_error(vec![hugr_int_ty]);
+        let sum_ty = self.0.typing_session().llvm_sum_type(hugr_sum_ty)?;
+
+        emit_custom_unary_op(self.0, args, |ctx, arg, _| {
+            // Build fabs intrinsic
+            let fabs_intr = Intrinsic::find("llvm.fabs.f64").expect("Couldn't find fabs intrinsic");
+            let fabs = fabs_intr
+                .get_declaration(
+                    ctx.get_current_module(),
+                    &[ctx.iw_context().f64_type().as_basic_type_enum()],
+                )
+                .ok_or(anyhow!("TODO"))?;
+
+            // We have to check if the conversion will work, so we
+            // make the maximum int and convert to a float, then compare
+            // with the function input.
+            let int_max = if signed {
+                int_ty.const_int(int_max_value_s, false)
+            } else {
+                int_ty.const_int(int_max_value_u, false)
+            };
+
+            let flt_int_max = if signed {
+                let abs_name = &format!("llvm.abs.i{}", width);
+                let abs_intr = Intrinsic::find(abs_name).expect("Couldn't find int abs intrinsic");
+                let abs = abs_intr
+                    .get_declaration(ctx.get_current_module(), &[int_ty.as_basic_type_enum()])
+                    .unwrap();
+
+                let const_false = ctx
+                    .iw_context()
+                    .bool_type()
+                    .const_zero()
+                    .as_basic_value_enum();
+                let abs_call = ctx
+                    .builder()
+                    .build_call(abs, &[int_max.into(), const_false.into()], "max_int_pos")?
+                    .as_any_value_enum()
+                    .into_int_value();
+
+                ctx.builder().build_signed_int_to_float(
+                    abs_call,
+                    ctx.iw_context().f64_type(),
+                    "max_flt_pos",
+                )
```

Comment:
Why bother to create an int constant and then cast it to float? Isn't it easier to cast `int_max_value_s` to a float in Rust and then emit a float constant?

Reply:
Oh yes, that works! Thanks!